### ISSUE TITLE
Add swagger documentation for API

### ISSUE THREAD

**Issue by dipeshdulal** [dipeshdulal/clean-gin#25]
Swagger Documentation for `clean-gin`

**Comment by Fimeo** [comment]
In order to continue the template with a swagger documentation. I propose to use an OpenApi v3 interface contract in json format in an api folder: `swagger.json`.

Moreover, it could be interesting to add a schema validator like https://github.com/getkin/kin-openapi on the requests made on the server through the creation of a middleware.

**Comment by dipeshdulal** [comment]
@Fimeo 
Thank you. Didn't know about `kin-openapi` project. 
Looks interesting and useful. 

So, `kin-openapi` can validate request inputs and response output from the openapi schema as well or just validate schema only? 🤔

**Comment by Fimeo** [comment]
Yes kin openapi takes the request and validates all schema against the
openapi definition.

It was really usefull for standard validation and provides low code and
easy validation.

You can validate the input request and if an error was found, cancel the
request with appropriate error message.

Le jeu. 23 févr. 2023 à 06:25, Dipesh Dulal ***@***.***> a
écrit :

**Comment by dipeshdulal** [comment]
Wow. 🤩 
This really looks useful. 
I will try to integrate with our architecture soon. 🏃 💨
